src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -193,6 +200,12 @@ public virtual void Process (LinkContext context)
 
 		void Initialize ()
 		{
+			// Pre-load corelib so that its XML gets processed first. This is necessary because the
+			// corelib atttribute XML can contain modifications to other assemblies.
```

Comment:
@vitek-karas since marking is separated from attribute XML processing, marking corelib here would only "happen to" process the attribute XML (when we mark some attribute, to check whether it should be removed). In a world where corelib didn't have any attributes, marking it wouldn't process the attribute XML.

So I think it's better to explicitly run just the attribute processing here, and let the normal mark logic take care of marking. I'll add a comment so that it's more obvious what this is for.